extism.go

```diff
@@ -276,6 +286,12 @@ func (m *Manifest) UnmarshalJSON(data []byte) error {
 
 // Close closes the plugin by freeing the underlying resources.
 func (p *Plugin) Close() error {
+	// TODO: Since we don't start the adapter in the ctor, should we stop it here?
+	if p.Adapter != nil {
```

Comment:
I prefer if the plugin starts and stops the adapter on its own, but since we cant currently start it (see the comment on `Plugin.Adapter`), should we stop the adapter?

Reply:
let me think about this for a little bit.. ideally we can share the adapter among plugins vs. an adapter per plugin. still looking at the implementation here though. 

each plugin should start and stop its own trace context -- but I think the adapter should be something the host program maintains.

Reply:
I see, in that case it shouldn't stop the adapter either. Thanks for the clarification